proxy/src/control/destination/mod.rs

```diff
@@ -1,5 +1,30 @@
+//! A client for the controller's Destination service.
+//!
+//! This client is split into two primary components: A `Resolver`, that routers use to
+//! initiate service discovery for a given name, and a `background::Process` that
+//! satisfies these resolution requests. These components are separated by a channel so
+//! that the thread responsible for proxying data need not also do this administrative
+//! work of communicating with the control plane.
+//!
+//! The number of active resolutions is not currently bounded by this module. Instead, we
+//! trust that callers of `Resolver` enforce such a constraint (for example, via
+//! `conduit_proxy_router`'s LRU cache).
```

Comment:
Additionally, users of this module must ensure they consume the notifications sent so that the notification message buffers don't grow without bounds.

(It might make more sense to do this a different way; e.g. send a "there is an update" message to the consumers, and have the consumers query the current state directly from the destination module under a lock when it is ready to process the update. This way, the inter-thread communication channel would be bounded to one message.)